zerver/models/streams.py

```diff
@@ -385,3 +385,19 @@ def to_dict(self) -> Dict[str, Any]:
 
 def get_default_stream_groups(realm: Realm) -> QuerySet[DefaultStreamGroup]:
     return DefaultStreamGroup.objects.filter(realm=realm)
+
+
+class StreamTopic(models.Model):
+    MAX_NAME_LENGTH = 60
+
+    realm = models.ForeignKey(Realm, on_delete=CASCADE)
+    stream_id = models.ForeignKey(Stream, db_index=True, on_delete=CASCADE)
+    topic_name = models.CharField(max_length=MAX_NAME_LENGTH)
```

Comment:
We can probably call this field `name` given that it'll be usually accessed via `topic.name`.